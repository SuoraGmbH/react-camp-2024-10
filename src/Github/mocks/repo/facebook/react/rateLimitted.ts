import { http, HttpResponse } from "msw";

export const githubRepoFacebookReactRateLimitedMockHandler = http.get(
  "https://api.github.com/repos/facebook/react",
  () => {
    return HttpResponse.json(
      {
        error: "Rate limited",
      },
      { status: 403 },
    );
  },
);
