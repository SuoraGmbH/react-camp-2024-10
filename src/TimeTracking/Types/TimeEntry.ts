import { z } from "zod";

export const timeEntrySchema = z.object({
  id: z.string(),
  comment: z.string(),
  start: z.coerce.date(),
  end: z.coerce.date(),
});

type TimeEntry = z.infer<typeof timeEntrySchema>;
export default TimeEntry;
