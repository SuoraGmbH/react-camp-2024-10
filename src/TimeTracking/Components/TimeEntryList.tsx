import React from "react";
import TimeEntry from "../Types/TimeEntry.ts";
import TimeEntryView from "./TimeEntryView.tsx";

interface Props {
  timeEntries: TimeEntry[];
}

const TimeEntryList: React.FunctionComponent<Props> = ({ timeEntries }) => {
  return (
    <>
      {timeEntries.map((timeEntry) => (
        <TimeEntryView key={timeEntry.id} timeEntry={timeEntry} />
      ))}
    </>
  );
};

export default TimeEntryList;
