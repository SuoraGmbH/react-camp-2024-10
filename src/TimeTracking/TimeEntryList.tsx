import React from "react";
import TimeEntry from "../domain/TimeEntry.ts";
import TimeEntryView from "./TimeEntryView.tsx";

interface Props {
  timeEntries: TimeEntry[];
}

const TimeEntryList: React.FunctionComponent<Props> = ({ timeEntries }) => {
  return (
    <>
      {timeEntries.map((timeEntry) => (
        <TimeEntryView key={timeEntry.id} timeEntry={timeEntry} />
      ))}
    </>
  );
};

export default TimeEntryList;
