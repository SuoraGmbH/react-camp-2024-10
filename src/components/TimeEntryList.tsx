import React from "react";
import TimeEntry from "../domain/TimeEntry.ts";

const TimeEntryList: React.FunctionComponent = () => {
  const timeEntries: TimeEntry[] = [
    {
      id: "ecf06450-2cf7-4ce9-adf6-7263e3027422",
      comment: "React lernen",
      start: new Date(),
      end: new Date(),
    },
    {
      id: "ecf06450-2cf7-4ce9-adf8-7263e3027422",
      comment: "React testing lernen",
      start: new Date(),
      end: new Date(),
    },
    {
      id: "ecf06450-2cf7-3ce9-adf6-7263e3027422",
      comment: "React Query lernen",
      start: new Date(),
      end: new Date(),
    },
  ];

  return (
    <ul>
      {timeEntries.map((timeEntry) => {
        return <li key={timeEntry.id}>{timeEntry.comment}</li>;
      })}
    </ul>
  );
};

export default TimeEntryList;
