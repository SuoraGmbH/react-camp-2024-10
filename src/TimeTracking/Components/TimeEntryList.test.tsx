import { render, screen } from "@testing-library/react";
import TimeEntryList from "./TimeEntryList.tsx";
import TimeEntry from "../Types/TimeEntry.ts";

const timeEntryA: TimeEntry = {
  id: "a0da97db-f4fa-4ced-869b-fa19b0040cb6",
  comment: "Ein erster Test",
  start: new Date("2024-10-09T11:59:27.122Z"),
  end: new Date("2024-10-09T11:59:27.122Z"),
};

const timeEntryB: TimeEntry = {
  id: "a0da97db-f4fa-4ced-869b-fa19b0540cb6",
  comment: "Ein zweiter Test",
  start: new Date(),
  end: new Date(),
};

describe("<TimeEntryList />", () => {
  test("renders without crashing", () => {
    render(<TimeEntryList timeEntries={[]} />);
  });

  test("it display one time entry if we pass one time entry", () => {
    render(<TimeEntryList timeEntries={[timeEntryA]} />);

    expect(screen.getByText(/ein erster test/i)).toBeVisible();
  });

  test("it displays both time entries if we pass two time entries", () => {
    render(<TimeEntryList timeEntries={[timeEntryA, timeEntryB]} />);

    expect(screen.getByText(/ein erster test/i)).toBeVisible();
    expect(screen.getByText(/ein zweiter test/i)).toBeVisible();
  });

  test("it renders html according to snapshot", () => {
    const { container } = render(
      <TimeEntryList timeEntries={[timeEntryA, timeEntryB]} />,
    );

    expect(container).toMatchSnapshot();
  });

  test("it renders text according to snapshot", () => {
    const { container } = render(
      <TimeEntryList timeEntries={[timeEntryA, timeEntryB]} />,
    );

    expect(container.textContent).toMatchInlineSnapshot(
      `"Ein erster TestEin zweiter Test"`,
    );

    expect(timeEntryA).toMatchInlineSnapshot(`
      {
        "comment": "Ein erster Test",
        "end": 2024-10-09T11:59:27.122Z,
        "id": "a0da97db-f4fa-4ced-869b-fa19b0040cb6",
        "start": 2024-10-09T11:59:27.122Z,
      }
    `);
  });
});
