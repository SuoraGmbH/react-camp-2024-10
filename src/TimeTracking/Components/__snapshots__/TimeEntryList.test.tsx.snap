// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TimeEntryList /> > it renders html according to snapshot 1`] = `
<div>
  <div>
    Ein erster Test
  </div>
  <div>
    Ein zweiter Test
  </div>
</div>
`;
